### PR TITLE
Use first 8 chars from first key as prefix for transaction log file

### PR DESCRIPTION
**Pull request by mankyKitty**

### CODE REVIEW

**Comment by mankyKitty** [comment]
Running a test ghcjs build to ensure the use of filepath hasn't broken anything.

File: frontend/src/Frontend/UI/Dialogs/TxLogs.hs [hunk 1]
```diff
@@ -9,7 +9,7 @@ module Frontend.UI.Dialogs.TxLogs
   ( uiTxLogs
   ) where
 
-import Control.Lens (iforM_)
+import Control.Lens -- (iforM_, (^.), _head)
```
Comment (alexfmpe) [inline frontend/src/Frontend/UI/Dialogs/TxLogs.hs]: Leftovers

File: frontend/src/Frontend/UI/Dialogs/TxLogs.hs [hunk 2]
```diff
@@ -97,20 +104,28 @@ uiTxLogs fileFFI _onExtClose = do
...
-        liftIO (Api._transactionLogger_exportFile txlog) <$ onClick
+        liftIO (Api._transactionLogger_exportFile txlog pk) <$ onClick
 
       onDeliveredFileOk <- _fileFFI_deliverFile fileFFI onContentsReady
 
       status <- holdDyn "fa-download" $ leftmost
-        [ "tx-log-export-fail fa-times" <$ onFileErr
-        , "tx-log-export-success fa-check" <$ ffilter isRight onDeliveredFileOk
+        [ "tx-log-export-fail fa-times" <$ traceEvent "file err" onFileErr
+        , "tx-log-export-success fa-check" <$ ffilter isRight (traceEvent "delivered file" onDeliveredFileOk)
```
Comment (alexfmpe) [inline frontend/src/Frontend/UI/Dialogs/TxLogs.hs]: Leftovers

File: frontend/src/Frontend/UI/Dialogs/TxLogs.hs [hunk 3]
```diff
@@ -97,20 +104,28 @@ uiTxLogs fileFFI _onExtClose = do
           td "tx-log-row__request-key" $ elClass "span" "request-key-text" $ text
             $ Pact.hashToText $ Pact.unRequestKey $ Api._commandLog_requestKey cmdLog
 
-  _ <- modalFooter $
-    exportButton txLogger
+  -- Both export wallet and txnlog export use the first 8 characters of the first
+  -- key in the wallet. Given that key is deterministic, could we simply generate it
+  -- rather than either loading keys or requiring them to exist in the wallet first?
```
Comment (alexfmpe) [inline frontend/src/Frontend/UI/Dialogs/TxLogs.hs]: Let's figure this out before merging

Comment (mankyKitty) [reply frontend/src/Frontend/UI/Dialogs/TxLogs.hs]: After discussions with Kadena, key prefix is generated instead of looked up.

**Pull request merged**